WcaOnRails/app/controllers/users_controller.rb

```diff
@@ -170,8 +170,16 @@ def update
     old_confirmation_sent_at = @user.confirmation_sent_at
     if @user.update(user_params)
       if @user.saved_change_to_delegate_status
-        # TODO: See https://github.com/thewca/worldcubeassociation.org/issues/2969.
-        DelegateStatusChangeMailer.notify_board_and_assistants_of_delegate_status_change(@user, current_user).deliver_now
+        if @user.delegate_status
+          if @user.delegate_status == "senior_delegate"
+            @user_senior_delegate = @user
+          else
+            @user_senior_delegate = @user.senior_delegate
+          end
+        else
+          @user_senior_delegate = User.find_by_id!(@user.senior_delegate_id_before_last_save)
```

Comment:
This _should_ be the same (haven't tested, but pretty sure about the standard ActiveRecord APIs) as `User.find!` (where any model's `find` method takes the model's primary key as search criterion by default)

Reply:
Done.